### PR TITLE
Superagent: Added support for multiple timeouts

### LINKED ISSUE #16756: @types/superagent missing multiple timeouts
Superagent is only accepting one timeout on requests. According to the [docs](https://visionmedia.github.io/superagent/#timeouts) it should accept response, read and deadline timeout.

### PR DESCRIPTION
**Pull request by ricmello**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://visionmedia.github.io/superagent/#timeouts>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Closes #16756

### CODE REVIEW

File: types/superagent/index.d.ts [hunk 1]
```diff
@@ -102,7 +103,7 @@ declare namespace request {
       send(): this;
       set(field: string, val: string): this;
       set(field: Object): this;
-      timeout(ms: number): this;
+      timeout(ms: number | Object | { response: number, read: number, deadline: number }): this;
```
Comment (mhegazy) [inline types/superagent/index.d.ts]: Did you mean 'object'?

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by ricardo-mello** [comment]
@mhegazy I copied the `Object` param from superagent signature, but it's really not necessary. Updated the branch.

File: types/superagent/index.d.ts (hunk 1 above)
Comment (ricardo-mello) [inline types/superagent/index.d.ts]: @mhegazy I copied the Object param from superagent signature, but it's really not necessary. Updated the branch.

**Comment by mxl** [comment]
@ricardo-mello superagent v2.x [does not have](https://github.com/visionmedia/superagent/blob/v2.3.0/lib/request-base.js#L55) these options in `req.timeout()` method. I'm preparing PR with superagent v3 definitions and suggest to rebase your changes on master after merging that PR.

**Comment by mxl** [comment]
@ricardo-mello @mhegazy Please look at #16920

**Comment by ricardo-mello** [comment]
@mxl you're right. I forgot to change the version, but if you're working on all the features of the new version, this PR is unnecessary.

**Pull request closed without merging**